### PR TITLE
Improve utils api doc

### PR DESCRIPTION
**Pull request by minggo**
Re: https://github.com/cocos/cocos-engine/issues/13888

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/utils/id-generator.ts [hunk 1]
```diff
@@ -45,26 +45,42 @@ export class IDGenerator {
     */
     public static global = new IDGenerator('global');
 
+    /**
+     * @en A number to record current id. It may increase when invoke `getNewId()`. Should use
+     * `getNewId()` to get a unique id.
+     * @zh 记录当前 id 值。大概调用 `getNewId()` 是，它可能被加1。应该使用 `getNewId()` 获取唯一的 id。
```
Comment (SantyWang) [inline cocos/core/utils/id-generator.ts]: typo

File: cocos/core/utils/js-typed.ts [hunk 2]
```diff
@@ -77,14 +93,15 @@ export function isEmptyObject (obj: any) {
...
- * @param [enumerable=false]
+ * @en A helper function to add a property to an object, or modifies attributes
+ * of an existing property. The property is configurable.
+ * @zh 为对象添加属性或者修改已有属性的辅助函数。该属性是可配置的。
+ * @param object @en The object to add or modify attributes. @zh 要添加或者修改属性的对象。
+ * @param propertyName @en The property name to add or modify. @zh 要添加或修改的属性名。
+ * @param value_ @en The property value to add or new value to replace a existing attribute.
+ * @zh 要添加的属性值，或者取代现有属性的新值。
+ * @param writable @en If the property is writable. @zh 属性是否可写。
+ * @param enumerable @en If the property is enumerable. @zh 属性是否可列举。
```
Comment (SantyWang) [inline cocos/core/utils/js-typed.ts]: 列举 -> 枚举

File: cocos/core/utils/js-typed.ts [hunk 3]
```diff
@@ -441,14 +489,15 @@ export function mixin (object?: Record<string | number, any>, ...sources: any[])
 
 /**
  * @en
- * Derive the class from the supplied base class.
- * Both classes are just native javascript constructors, not created by `Class`, so
- * usually you will want to inherit using [[CCClass]] instead.
+ * Makes a class inherit from the supplied base class.
+ * Both classes should be native javascript constructors, not created by `Class`, so
+ * usually you will want to inherit from [[CCClass]] instead.
```
Comment (SantyWang) [inline cocos/core/utils/js-typed.ts]: This description is not correct, it's not inheriting from ccclass, it's using ccclass.

File: cocos/core/utils/js-typed.ts [hunk 4]
```diff
@@ -479,28 +528,28 @@ export function extend (cls: Function, base: Function) {
...
- * 判断一类型是否是另一类型的子类或本身
- * @param subclass sub class to be checked
- * @param superclass super class to be checked
- * @return whether subclass is child of superclass
+ * Checks whether a class is child of another class, or is the same type as another class.
+ * @zh 判断一类型是否是另一类型的子类或本身。
+ * @param subclass @en Sub class to check. @zh 子类类型。
+ * @param superclass @en Super class to check. @zh 父类类型。
+ * @return @en True if sub class is child of super class, or they are the same type. False else.
+ * 如果子类类型是父类类型的子类，或者二者是相同类型，那么返回 true，否则返回 false。
```
Comment (SantyWang) [inline cocos/core/utils/js-typed.ts]: miss `@zh`

File: cocos/core/utils/js-typed.ts [hunk 5]
```diff
@@ -603,13 +652,10 @@ export const _setClassId = setup('__cid__', _idToClass, false);
...
- * Register the class by specified name manually
- * @zh
- * 通过指定的名称手动注册类型
- * @method setClassName
- * @param className
- * @param constructor
+ * @en Registers a class by specified name manually.
+ * @zh 通过指定的名称手动注册类型
+ * @param className @en Class name to register. @zh 注册的类名。
+ * @param constructor @ Constructor to register. @zh 注册的构造函数。
```
Comment (SantyWang) [inline cocos/core/utils/js-typed.ts]: typo

Comment (minggo) [reply cocos/core/utils/js-typed.ts]: good catch

File: cocos/core/utils/id-generator.ts (hunk 1 above)
Comment (minggo) [reply cocos/core/utils/id-generator.ts]: fixed

File: cocos/core/utils/js-typed.ts (hunk 2 above)
Comment (minggo) [reply cocos/core/utils/js-typed.ts]: fixed

File: cocos/core/utils/js-typed.ts (hunk 3 above)
Comment (minggo) [reply cocos/core/utils/js-typed.ts]: fixed

File: cocos/core/utils/js-typed.ts (hunk 4 above)
Comment (minggo) [reply cocos/core/utils/js-typed.ts]: fixed

File: cocos/core/utils/js-typed.ts (hunk 5 above)
Comment (minggo) [reply cocos/core/utils/js-typed.ts]: fixed

**Review by SantyWang — APPROVED** [review]

**Pull request merged**